### DISCUSSION TITLE
Please consider normalizing the data

### DISCUSSION THREAD

**Discussion by jonhassall** [spatie/laravel-activitylog#1296] [Ideas] [answered]
Please consider normalizing the data, instead of repeating strings in the log_name, description, subject_type, event and causer_type columns.

Normalizing will improve efficiency, reduce potential errors, and make the data easier to query and analyze.

**Comment by Gummibeer** [comment, marked as answer]
It's all normalized - all have a different meaning and also can have different values. No column is forced to have the same value as another.
